infrastructure/execution/security.py

```
@@ -0,0 +1,244 @@
+    def _safe_import(self, name, globals=None, locals=None, fromlist=(), level=0):
+        """Safely import allowed modules only."""
+
+        if level != 0:
+            raise ImportError("Relative imports are not allowed")
+
+        allowed_names = set(self.allowed_modules.keys())
+        allowed_roots = {module.split(".")[0] for module in allowed_names}
+
+        root_name = name.split(".")[0]
+        if name not in allowed_names and root_name not in allowed_roots:
+            raise ImportError(f"Import of module '{name}' is not allowed")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)